backend/session 4/app.py

```diff
@@ -0,0 +1,334 @@
+        db.session.commit()
+        return jsonify("Task Deleted")
+    return jsonify("Not Found"), 404
+
+
+@app.route("/edit_task", methods=["PUT"])
+@jwt_required()
+def edit_tasks():
+    json = request.get_json()
+    current_user_email = get_jwt_identity()
+    user = User.query.filter_by(email=current_user_email).first()
+    your_work = json["text"]
+    edit_task = Task.query.filter_by( user_id=user.id).first()
+    edit_task.your_work = your_work
+    db.session.commit()
+    return jsonify("edited")
+
+
+@app.route("/view_tasks", methods=["GET"])
+@jwt_required()
+def view_tasks():
+    current_user_email = get_jwt_identity()
+    user = User.query.filter_by(email=current_user_email).first()
+    view_all_tasks = Task.query.filter_by( user_id=user.id)
+    tasks = [[ task.title,task.your_work] for task in view_all_tasks]
+    return jsonify(tasks)
+
+
+@app.route("/completed_tasks", methods=["PUT"])
+@jwt_required()
+def task_completed():
+    json = request.get_json()
+    current_user_email = get_jwt_identity()
+    user = User.query.filter_by(email=current_user_email).first()
+    task = Task.query.filter_by(title = json["title"], user_id=user.id).first()
+    if task:
+        task.completed = True
+        db.session.commit()
+        return jsonify("Task marked as completed")
+    return jsonify("task not found"), 404
+
+
+@app.route("/take_note", methods=["POST"])
+@jwt_required()
+def taking_notes():
+    json = request.get_json()
+    current_user_email = get_jwt_identity()
+    user = User.query.filter_by(email=current_user_email).first()
+    note = Note(
+        your_note=json["your_note"],
+        secret=json["secret"],
+        title_notes=json["title_notes"],
+        user_id=user.id,
+    )
+    db.session.add(note)
+    db.session.commit()
+    return jsonify("Note added")
+
+
+@app.route("/edit_note", methods=["PUT"])
+@jwt_required()
+def edit_note():
+    json = request.get_json()
+    current_user_email = get_jwt_identity()
+    user = User.query.filter_by(email=current_user_email).first()
+    title_notes = json["title_notes"]
+    note_edited = Note.query.filter_by(title_notes=title_notes, user_id=user.id).first()
+    note_edited.your_note = json["your_note"]
+    note_edited.secret = json["secret"]
+    db.session.commit()
+    return jsonify("edited"), 200
+
+
+@app.route("/get_note_name", methods=["GET"])
+@jwt_required()
+def view():
+    json = request.get_json()
+    current_user_email = get_jwt_identity()
+    user = User.query.filter_by(email=current_user_email).first()
+    noted = Note.query.filter_by(secret=False, user_id=user.id)
+    notes = [note.your_note for note in noted]
+    return jsonify({"your_note": notes})
+
+
+@app.route("/remove_note", methods=["DELETE"])
+@jwt_required()
+def remove_note():
+    current_user_email = get_jwt_identity()
+    user = User.query.filter_by(email=current_user_email).first()
+    note = Note.query.filter_by(user_id=user.id).first()
```

Comment:
you're deleting all user notes!
which is wrong
you have to take from frontend note title or sth to filter by it